docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,428 @@
+---
+title: dApp Tutorial
+description: Build a complete dApp on Kadena
+---
+
+import PageRef from '@components/PageRef'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Complete dApp Tutorial
+
+Build a complete on-chain voting dApp
+
+---
+
+## Overview
+
+In this tutorial you will build a complete dApp, smart contracts and frontend that allows users to submit a vote. You will also deploy your own gas station that will be used to pay for the gas needed to interact with your dApp.
+
+## Environment Setup
+
+### Install Pact
+
+**Using Brew**
+
+```clojure
+brew update
+brew install kadena-io/pact/pact
+```
+**Build from source**
+- follow the instructions [here](https://github.com/kadena-io/pact#building-from-source)
+
+**Verifying Installation**
+
+Type `pact` in a terminal and try out some commands:
+
+```clojure
+$ pact
+pact> (+ 1 2)
+3
+pact> (+ "hello, " "world")
+"hello, world"
+pact> (< 0 1)
+true
+```
+
+### Create Project Structure
+
+Copy paste the instructions below to create a basic project structure:
+
+```clojure
+mkdir my-pact-project && cd my-pact-project
+mkdir pact
+mkdir app
+```
+
+### Pact Http Server
+
+Create a `config.yaml` file for pact http server. We will later use it simulate blockchains transactions locally.
+
+For more details about pact http server check <a href="/blog/deploy-to-a-local-server">this</a> tutorial.
+
+```clojure
+touch config.yaml
+```
+Copy the code below and save the file:
+
+```clojure
+# Config file for pact http server. Launch with `pact -s config.yaml`
+
+# HTTP server port
+port: 8080
+
+# directory for HTTP logs
+logDir: log
```

Comment:
Need to `mkdir log` too or else starting the server doesn't work

Reply:
@jmininger I moved the Pact Server setup to the Pact Server section since that's where it's used and added `mkdir`.